website/tables/azure/azure_network_private_link_services.md

```diff
@@ -0,0 +1,25 @@
+# Table: azure_network_private_link_services
+
+This table shows data for Azure Network Private Link Services.
+
+https://learn.microsoft.com/en-us/rest/api/virtualnetwork/private-link-services/list-by-subscription?tabs=HTTP#privatelinkservice
+
+The primary key for this table is **id**.
+
+## Columns
+
+| Name          | Type          |
+| ------------- | ------------- |
+|_cq_source_name|String|
+|_cq_sync_time|Timestamp|
+|_cq_id|UUID|
+|_cq_parent_id|UUID|
+|subscription_id|String|
```

Comment:
Should the subscription be part of the PK maybe, to be on the safe side? 🤔

Reply:
We usually don't add subscription as a PK in Azure as it's included in the ID already, see example data in https://learn.microsoft.com/en-us/rest/api/virtualnetwork/private-link-services/list-by-subscription?tabs=HTTP:

```json
{
  "value": [
    {
      "name": "testPls1",
      "id": "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/privateLinkServices/testPls1",
      "type": "Microsoft.Network/privateLinkServices",
      "location": "eastus",
      "properties": {
        "provisioningState": "Succeeded",
        "visibility": {
          "subscriptions": [
            "subscription1",
            "subscription2",
            "subscription3"
          ]
        },
        "autoApproval": {
          "subscriptions": [
            "subscription1",
            "subscription2"
          ]
        },
        "fqdns": [
          "fqdn1",
          "fqdn2",
          "fqdn3"
        ],
        "alias": "ContosoService.{guid}.azure.privatelinkservice",
        "loadBalancerFrontendIpConfigurations": [
          {
            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/loadBalancers/lb/frontendIPConfigurations/fe-lb1"
          }
        ],
        "ipConfigurations": [
          {
            "name": "fe-lb1",
            "properties": {
              "privateIPAddress": "10.0.1.4",
              "privateIPAllocationMethod": "Static",
              "privateIPAddressVersion": "IPv4",
              "subnet": {
                "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.Network/virtualNetworks/vnetlb/subnets/subnetlb1"
              }
            }
          }
        ],
        "privateEndpointConnections": [
          {
            "name": "pec1",
            "properties": {
              "privateEndpoint": {
                "id": "/subscriptions/subId/resourceGroups/rg1/providers/Microsoft.Network/privateEndpoints/testPe1"
              },
              "privateLinkServiceConnectionState": {
                "status": "Approved",
                "description": "approved it for some reason."
              }
            }
          }
        ],
        "networkInterfaces": [
          {
            "id": "/subscriptions/subId/resourceGroups/rg1/provders/Microsoft.Network/networkInterfaces/testPls1.nic.abcd1234"
          }
        ]
      }
    },
    {
      "name": "testPls2",
      "id": "/subscriptions/subId/resourceGroups/rg2/providers/Microsoft.Network/privateLinkServices/testPls2",
      "type": "Microsoft.Network/privateLinkServices",
      "location": "eastus",
      "properties": {
        "provisioningState": "Succeeded",
        "visibility": {
          "subscriptions": [
            "subscription1",
            "subscription2"
          ]
        },
        "autoApproval": {
          "subscriptions": [
            "subscription1",
            "subscription2"
          ]
        },
        "fqdns": [
          "fqdn1",
          "fqdn2"
        ],
        "alias": "ContosoService.{guid}.azure.privatelinkservice",
        "loadBalancerFrontendIpConfigurations": [
          {
            "id": "/subscriptions/subid/resourceGroups/rg2/providers/Microsoft.Network/loadBalancers/lb/frontendIPConfigurations/fe-lb2"
          }
        ],
        "ipConfigurations": [
          {
            "name": "fe-lb2",
            "properties": {
              "privateIPAddress": "10.0.1.5",
              "privateIPAllocationMethod": "Static",
              "privateIPAddressVersion": "IPv4",
              "subnet": {
                "id": "/subscriptions/subid/resourceGroups/rg2/providers/Microsoft.Network/virtualNetworks/vnetlb/subnets/subnetlb2"
              }
            }
          }
        ],
        "privateEndpointConnections": [
          {
            "name": "pec1",
            "properties": {
              "privateEndpoint": {
                "id": "/subscriptions/subId/resourceGroups/rg2/providers/Microsoft.Network/privateEndpoints/testPe2"
              },
              "privateLinkServiceConnectionState": {
                "status": "Approved",
                "description": "approved it for some reason."
              }
            }
          }
        ],
        "networkInterfaces": [
          {
            "id": "/subscriptions/subId/resourceGroups/rg2/provders/Microsoft.Network/networkInterfaces/testPls2.nic.efgh5678"
          }
        ]
      }
    }
  ]
}
```

Reply:
Ah okay cool, all good then, thanks! 👍